### DISCUSSION TITLE
重大建议：集中式订阅库，自动订阅

### DISCUSSION THREAD

**Discussion by zenglian** [v2fly/v2ray-core#1787] [Ideas]
### 问题：
现在的订阅方法是：
1. 上网搜索， 复制URL手工添加
2. 手工测试速度，选择最快服务器。
3. 如果不行了，重复到第1步

### 想法
把上述过程自动化。
1. 构建一个或数个“官方”的集中式的订阅库
2. 用户在订阅库server端主动添加自己收集的proxy list，不再以零散网页的方式提供
3. 客户端软件在用户允许的情况自动收集proxy地址上报到server
4. 对于提供者给予某种形式的奖励
5. 订阅库server端自动过滤重复的、不能连接的proxy，清理长期不能连接的proxy
6. 客户端软件启动时自动连接和查询订阅库，并自动切换性能较佳的proxy（可选）
7. 订阅库server端自动做负载均衡，根据客户端位置自动返回性能较好的若干个代理
8. 需要解决订阅库本身被墙的问题

**Comment by cupen** [comment]
机器、流量、以及人力都是要钱的。你愿意为这个功能付钱么？愿意的话，梯子商人会替你解决。
此外，集中暴露理论上很容易被一锅端，都不用探测了，直接拿到枪毙名单。
